staging/src/k8s.io/apiserver/pkg/cel/openapi/schemas.go

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi
+
+import (
+	"strings"
+	"time"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+
+	apiservercel "k8s.io/apiserver/pkg/cel"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+func SchemaBy(f func(s *spec.Schema)) *spec.Schema {
+	s := new(spec.Schema)
+	f(s)
+	return s
+}
```

Comment:
this doesn't look used anywhere, unexport it?

actually, I'm not sure what this is getting us... it looks like all the uses pass in a closure... so instead of this:
```go
&spec.Schema{
  Type: []string{"object"},
  AdditionalProperties: &spec.SchemaOrBool{
    Schema: &spec.Schema{
      Type: []string{"string"}
    },
  },
},
```
we can write this:
```go
by(func(s *spec.Schema) {
  s.Type = []string{"object"}
  s.AdditionalProperties = &spec.SchemaOrBool{
    Schema: by(func(s *spec.Schema) {
      s.Type = []string{"string"}
    }),
  }
}),
```

those look... exactly the same length, and the straightforward struct declaration is easier to understand

should we drop the indirection to simplify the code?

Reply:
The former is actually this 
```go
&spec.Schema{
SchemaProps: spec.SchemaProps{
	Type: []string{"string"},
	AdditionalProperties: &spec.SchemaOrBool{
		Schema: &spec.Schema{
			SchemaProps: spec.SchemaProps{
				Type: []string{"string"},
			},
		},
	},
},
```
Unfortunately, nested object can be directly accessed but not directly constructed.

Reply:
hmm... anything with AdditionalProperties should be an object, right?

looks like there are construction helpers in `spec` so that `spec.MapProperty(spec.StringProperty())` is equivalent to this (and sets Allows:true correctly in SchemaOrBool)

```go
&spec.Schema{
	SchemaProps: spec.SchemaProps{
		Type: []string{"object"},
		AdditionalProperties: &spec.SchemaOrBool{
			Allows: true,
			Schema: &spec.Schema{
				SchemaProps: spec.SchemaProps{
					Type: []string{"string"},
				},
			},
		},
	},
}
```

Reply:
The fluent helpers do not cover every fields we require, especially extensions. A mixture of construct and helpers is no better than pure construct.

Reply:
fair enough, let's just not export it past this package to avoid picking up other callers

Reply:
This function saves only one layer of the human readers' reading time, I will go ahead to use the helpers when possible and direct construct otherwise.